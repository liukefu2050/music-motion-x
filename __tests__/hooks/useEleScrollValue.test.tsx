import { act, screen, fireEvent } from "@testing-library/react"
import { useCallback, useRef } from "react"
import useEleScrollValue from "../../src/hooks/useEleScrollValue"
import renderWithRoot from "../test-utils/renderWithRoot"

describe("useEleScrollValue", () => {
  it("worked when page scroll", () => {
    const fn = jest.fn()
    const formatter = (number: number) => {
      fn(number)
      return Number.isNaN(number) ? 0 : number
    }
    const App = ({ formatter }: { formatter: (num: number) => number }) => {
      const scrollContainerRef = useRef<HTMLDivElement | null>(null)
      const callScrollContainerRef = useCallback(
        () => scrollContainerRef.current!,
        [],
      )
      const headerOpacity = useEleScrollValue(callScrollContainerRef, formatter)

      return (
        <>
          <div style={{ opacity: headerOpacity }}>test</div>
          <div ref={scrollContainerRef} style={{ height: 800 }}>
            dedede
          </div>
        </>
      )
    }
    const { asFragment, rerender } = renderWithRoot(
      <App formatter={formatter} />,
    )
    const pageRef = screen.getAllByTestId("test-root")?.[0]

    // eslint-disable-next-line testing-library/no-unnecessary-act
    act(() => {
      fireEvent.scroll(pageRef, { target: { scrollTop: 300 } })
    })

    expect(fn).toBeCalled()
    expect(asFragment()).toMatchSnapshot()

    const fn1 = jest.fn()
    const formatter1 = (number: number) => {
      fn1(number)
      return Number.isNaN(number) ? 0 : number
    }

    rerender(<App formatter={formatter1} />)
    // eslint-disable-next-line testing-library/no-unnecessary-act
    act(() => {
      fireEvent.scroll(pageRef, { target: { scrollTop: 300 } })
    })
  })
})
