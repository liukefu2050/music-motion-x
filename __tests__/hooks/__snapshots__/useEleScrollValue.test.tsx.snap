// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useEleScrollValue worked when page scroll 1`] = `
<DocumentFragment>
  <div
    style="opacity: 1;"
  >
    test
  </div>
  <div
    style="height: 800px;"
  >
    dedede
  </div>
</DocumentFragment>
`;
